cli/actions/resource.go

```diff
@@ -0,0 +1,47 @@
+package actions
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+type ApplyArgs struct {
+	File string
+}
+
+type ResourceActions interface {
+	Apply(ctx context.Context, args ApplyArgs) error
+	List(ctx context.Context) error
+	Export(ctx context.Context, ID string) error
+	Delete(ctx context.Context, ID string) error
+}
+
+type resourceArgs struct {
+	logger  *zap.Logger
+	client  http.Client
+	request http.Request
+}
+
+type supportedResources string
+
+var (
+	Config supportedResources = "config"
+)
+
+func NewResourceActions(resourceType string, logger *zap.Logger, client http.Client, request http.Request) (ResourceActions, error) {
+	args := resourceArgs{
+		logger:  logger,
+		client:  client,
+		request: request,
+	}
+
+	switch resourceType {
+	case string(Config):
+		return NewConfigActions(args), nil
+	default:
+		return nil, fmt.Errorf("unsupported resource type: %s", resourceType)
+	}
```

Comment:
I think we can skip this switch, and use a more polymorphyc approach. Using a registry of some kind, you can have each resource register itself against a centralized registry, and have that registry get the requirec `ResourceActions` for you.

This will allow to easily implement new resource types, reduce the chance of "forgetting to add to the switch here or there" kind of errors, etc